lib/rubocop/cop/style/hash_except.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for usages of `Hash#reject` and `Hash#select` methods that can be
+      # replaced with `Hash#except` method.
+      #
+      # This cop should only be enabled on Ruby version 3.0 or higher.
+      # (`Hash#except` was added in Ruby 3.0.)
+      #
+      # For safe detection, it is limited to commonly used string and symbol comparisons
+      # when used `==`.
+      #
+      # @example
+      #
+      #   # bad
+      #   {foo: 1, bar: 2, baz: 3}.reject {|k, v| k == :bar }
+      #   {foo: 1, bar: 2, baz: 3}.select {|k, v| k != :bar }
+      #
+      #   # good
+      #   {foo: 1, bar: 2, baz: 3}.except(:bar)
+      #
+      class HashExcept < Base
+        include RangeHelp
+        extend TargetRubyVersion
+        extend AutoCorrector
+
+        minimum_target_ruby_version 3.0
+
+        MSG = 'Use `%<prefer>s` instead.'
+        RESTRICT_ON_SEND = %i[reject select].freeze
```

Comment:
I think you forgot about `filter`, `keep_if` and `delete_if`. I don't use those Hash methods often, but I recall there were a ton of aliases there.

Reply:
I added `filter` method. On the other hand, `delete_if` and` keep_if` methods  are not checked because they are incompatible with `except` method.

```ruby
# except
h = {foo: 1, bar: 2, baz: 3}
=> {:foo=>1, :bar=>2, :baz=>3}
h.except(:bar)
=> {:foo=>1, :baz=>3}
h
=> {:foo=>1, :bar=>2, :baz=>3}

# delete_if
h = {foo: 1, bar: 2, baz: 3}
=> {:foo=>1, :bar=>2, :baz=>3}
h.delete_if {|k, v| k == :bar }
=> {:foo=>1, :baz=>3}
h
=> {:foo=>1, :baz=>3}

# keep_if
h = {foo: 1, bar: 2, baz: 3}
=> {:foo=>1, :bar=>2, :baz=>3}
h.keep_if {|k, v| k != :bar }
=> {:foo=>1, :baz=>3}
h
=> {:foo=>1, :baz=>3}
```

I updated documentation and test on this.